types/react-dom/v15/test-utils/index.d.ts

```diff
@@ -156,8 +156,12 @@ export function renderIntoDocument<T extends Element>(
     element: DOMElement<any, T>): T;
 export function renderIntoDocument(
     element: SFCElement<any>): void;
-export function renderIntoDocument<T extends Component<any>>(
-    element: CElement<any, T>): T;
+// If we replace `P` with `any` in this overload, then some tests fail because
+// calls to `renderIntoDocument` choose the last overload on the
+// subtype-relation pass and get an undesirably broad return type.  Using `P`
+// allows this overload to match on the subtype-relation pass.
```

Comment:
does this change match ones in later react-dom versions?

Reply:
#34031 made dependent changes to `types/react-dom/v15/react-dom-tests.ts` (older) and `types/react-dom/test-utils/index.d.ts` (latest). 197675585a2fd685b831e8f2c74036eedd96a020 propagates the change to the older `test-utils/index.d.ts`.

0cad2799a298ff56d4358e07a32044ca2d13f03f was applied to the latest version (which the older `react-dom-tests.ts` was using) in #30748. The older version needs it, now that `strictNullChecks` is enabled.